caas/kubernetes/clientconfig/types.go

```diff
@@ -38,13 +40,13 @@ type CloudConfig struct {
 // Cluster_A, User_B: No new Cloud, new Credential for the cloud.
 
 // ClientConfigFunc is a function that returns a ClientConfig. Functions of this type should be available for each supported CAAS framework, e.g. Kubernetes.
-type ClientConfigFunc func() (*ClientConfig, error)
+type ClientConfigFunc func(io.Reader) (*ClientConfig, error)
 
 // NewClientConfigReader returns a function of type ClientConfigFunc to read the client config for a given cloud type.
 func NewClientConfigReader(cloudType string) (ClientConfigFunc, error) {
 	switch cloudType {
 	case "kubernetes":
-		return K8SClientConfig, nil
+		return NewClientConfig, nil
```

Comment:
Here is where we want to return a wrapper function to test if the supplied reader is nil and if it is nil, make one from a file, then pass that to NewClientConfig